File: rust/src/ui/settings_view.rs
// Settings form (plan section 4.3; options/defaults per SPEC 13.2):
// theme radio x3 (System default / Moonlit (light) / Moondark (dark)),
// interval radio x4 (1/5/10/30 min, default 5), autostart checkbox, Save.
// Save order: write JSON -> autostart -> theme -> reschedule timer (app.rs).

use crate::app::App;
use crate::theme::Palette;
use ratatui::layout::{Constraint, Layout, Rect};
use ratatui::style::{Modifier, Style};
use ratatui::text::{Line, Span};
use ratatui::widgets::{Block, Padding, Paragraph};
use ratatui::Frame;

const FOOTER: &str = "↑/↓ Move · ←/→ Change · Enter Save/Toggle · Esc Cancel";

const THEME_LABELS: [(&str, &str); 3] = [
    ("system", "System default"),
    ("light", "Moonlit (light)"),
    ("dark", "Moondark (dark)"),
];
const INTERVAL_LABELS: [(i64, &str); 4] = [(1, "1 min"), (5, "5 min"), (10, "10 min"), (30, "30 min")];

fn sel_style(selected: bool, p: &Palette) -> Style {
    if selected {
        Style::default().bg(p.button_hover).fg(p.text_primary)
    } else {
        Style::default().fg(p.text_primary)
    }
}

fn radio(selected_value: bool, row_selected: bool, label: &str, p: &Palette) -> Line<'static> {
    let mark = if selected_value { "(●)" } else { "( )" };
    let mark_style = if selected_value {
        Style::default().fg(p.accent)
    } else {
        Style::default().fg(p.text_secondary)
    };
    Line::from(vec![
        Span::styled(format!("{mark} "), mark_style),
        Span::styled(label.to_string(), sel_style(row_selected, p)),
    ])
}

pub fn draw(f: &mut Frame, app: &App, p: &Palette) {
    let chunks = Layout::vertical([
        Constraint::Length(3),
        Constraint::Min(10),
        Constraint::Length(1),
    ])
    .split(f.area());

    // Title bar (SPEC 13.1 copy)
    let title = Paragraph::new(Line::from(Span::styled(
        "Kimi Planbar TUI Settings",
        Style::default().fg(p.text_primary).add_modifier(Modifier::BOLD),
    )))
    .block(Block::default().padding(Padding::new(2, 0, 1, 0)));
    f.render_widget(title, chunks[0]);

    let inner = Rect {
        x: chunks[1].x,
        y: chunks[1].y + 1,
        width: chunks[1].width,
        height: chunks[1].height.saturating_sub(1),
    };

    let Some(draft) = &app.settings_draft else {
        return;
    };

    let heading = |text: &str| {
        Line::from(Span::styled(
            text.to_string(),
            Style::default()
                .fg(p.text_secondary)
                .add_modifier(Modifier::BOLD),
        ))
    };
    let mut lines: Vec<Line> = vec![
        heading("Theme"),
        Line::raw(""),
    ];
    for (value, label) in THEME_LABELS {
        lines.push(radio(draft.theme == value, app.settings_sel == 0, label, p));
    }
    lines.push(Line::raw(""));
    lines.push(heading("Refresh interval"));
    lines.push(Line::raw(""));
    // Interval pills on one row (SPEC 13.2 horizontal StackPanel)
    let mut pill_spans: Vec<Span> = Vec::new();
    for (value, label) in INTERVAL_LABELS {
        let active = draft.refresh_minutes == value;
        let style = if active {
            Style::default().fg(ratatui::style::Color::White).bg(p.accent)
        } else {
            Style::default().fg(p.text_primary).bg(p.button_bg)
        };
        let style = if app.settings_sel == 1 {
            style.add_modifier(Modifier::UNDERLINED)
        } else {
            style
        };
        pill_spans.push(Span::styled(format!(" {label} "), style));
        pill_spans.push(Span::raw(" "));
    }
    lines.push(Line::from(pill_spans));
    lines.push(Line::raw(""));
    // Autostart checkbox
    let check = if draft.auto_start { "[x]" } else { "[ ]" };
    let check_style = if draft.auto_start {
        Style::default().fg(p.accent)
    } else {
        Style::default().fg(p.text_secondary)
    };
    lines.push(Line::from(vec![
        Span::styled(format!("{check} "), check_style),
        Span::styled("Launch at Windows startup", sel_style(app.settings_sel == 2, p)),
    ]));
    lines.push(Line::raw(""));
    // Save button
    let save_style = if app.settings_sel == 3 {
        Style::default().fg(ratatui::style::Color::White).bg(p.accent)
    } else {
        Style::default().fg(p.text_primary).bg(p.button_bg)
    };
    lines.push(Line::from(Span::styled(" Save ", save_style)));

    f.render_widget(Paragraph::new(lines), Rect { ..inner });

    f.render_widget(
        Paragraph::new(FOOTER).style(Style::default().fg(p.text_secondary)),
        chunks[2],
    );
}

File: go/internal/tui/settingsview.go
// Settings form (SPEC 13.2): theme radio x3 (System default / Moonlit
// (light) / Moondark (dark)), interval pills x4 (1/5/10/30 min, default 5),
// autostart checkbox, Save. Save order — write JSON -> autostart -> theme ->
// reschedule timer — lives in app.go saveSettings, mirroring rust app.rs.
// Row layout translated from rust/src/ui/settings_view.rs.
package tui

import (
	"github.com/shawn-0106t/kimi-planbar-tui/go/internal/core"
)

const settingsFooter = "↑/↓ Move · ←/→ Change · Enter Save/Toggle · Esc Cancel"

type themeLabel struct{ value, label string }

var themeLabels = [3]themeLabel{
	{"system", "System default"},
	{"light", "Moonlit (light)"},
	{"dark", "Moondark (dark)"},
}

type intervalLabel struct {
	value int64
	label string
}

var intervalLabels = [4]intervalLabel{
	{1, "1 min"}, {5, "5 min"}, {10, "10 min"}, {30, "30 min"},
}

// selStyle highlights the row the cursor is on.
func selStyle(selected bool, p core.Palette) Span {
	if selected {
		return Span{Fg: p.TextPrimary, Bg: p.ButtonHover}
	}
	return Span{Fg: p.TextPrimary}
}

// radioLine is one radio row: "(●)" / "( )" + label.
func radioLine(selectedValue, rowSelected bool, label string, p core.Palette) Line {
	mark := "( )"
	markFg := p.TextSecondary
	if selectedValue {
		mark = "(●)"
		markFg = p.Accent
	}
	labelSpan := selStyle(rowSelected, p)
	labelSpan.Text = label
	return Line{Spans: []Span{
		{Text: mark + " ", Fg: markFg},
		labelSpan,
	}}
}

// headingLine is a section heading (bold secondary).
func headingLine(text string, p core.Palette) Line {
	s := styledSpan(text, p.TextSecondary)
	s.Bold = true
	return Line{Spans: []Span{s}}
}

// settingsView returns the content rows above the footer: blank, padded
// title, blank (3-row title block), blank (the inner-rect top offset in the
// Rust layout), then the form lines.
func settingsView(m *Model, p core.Palette) []Line {
	title := styledSpan("  Kimi Planbar TUI Settings", p.TextPrimary)
	title.Bold = true

	rows := []Line{
		blankOf(p),
		{Spans: []Span{title}},
		blankOf(p),
		blankOf(p),
	}

	if m.settingsDraft == nil {
		return rows
	}
	draft := m.settingsDraft

	rows = append(rows, headingLine("Theme", p), blankOf(p))
	for _, tl := range themeLabels {
		rows = append(rows, radioLine(draft.Theme == tl.value, m.settingsSel == 0, tl.label, p))
	}
	rows = append(rows, blankOf(p), headingLine("Refresh interval", p), blankOf(p))

	// Interval pills on one row (SPEC 13.2 horizontal StackPanel).
	var pillSpans []Span
	for _, il := range intervalLabels {
		pill := Span{Text: " " + il.label + " ", Bg: p.ButtonBg, Fg: p.TextPrimary}
		if draft.RefreshMinutes == il.value {
			pill.Fg = "#FFFFFF" // ratatui Color::White on the accent pill
			pill.Bg = p.Accent
		}
		if m.settingsSel == 1 {
			pill.Underline = true
		}
		pillSpans = append(pillSpans, pill, spanOf(" "))
	}
	rows = append(rows, Line{Spans: pillSpans}, blankOf(p))

	// Autostart checkbox.
	check := "[ ]"
	checkFg := p.TextSecondary
	if draft.AutoStart {
		check = "[x]"
		checkFg = p.Accent
	}
	labelSpan := selStyle(m.settingsSel == 2, p)
	labelSpan.Text = "Launch at Windows startup"
	rows = append(rows, Line{Spans: []Span{
		{Text: check + " ", Fg: checkFg},
		labelSpan,
	}}, blankOf(p))

	// Save button.
	save := Span{Text: " Save ", Bg: p.ButtonBg, Fg: p.TextPrimary}
	if m.settingsSel == 3 {
		save.Fg = "#FFFFFF"
		save.Bg = p.Accent
	}
	rows = append(rows, Line{Spans: []Span{save}})
	return rows
}

// cycle moves through the options wrapping in `dir` (rust app.rs cycle).
func cycleI64(options []int64, current int64, dir int) int64 {
	idx := 0
	for i, o := range options {
		if o == current {
			idx = i
			break
		}
	}
	return options[((idx+dir)%len(options)+len(options))%len(options)]
}

func cycleStr(options []string, current string, dir int) string {
	idx := 0
	for i, o := range options {
		if o == current {
			idx = i
			break
		}
	}
	return options[((idx+dir)%len(options)+len(options))%len(options)]
}

var themeValues = []string{"system", "light", "dark"}
var intervalValues = []int64{1, 5, 10, 30}

// settingsFooterRow exposes the footer text (used by app.go's frame builder).
func settingsFooterRow(p core.Palette) Line {
	return Line{Spans: []Span{styledSpan(settingsFooter, p.TextSecondary)}}
}
